### PR TITLE
Cleanup labels when PR gets closed

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary

Cleanup labels when a PR gets closed that don't make sense on a closed PR like `Setup Cloud Test Server` or `Do Not Merge`.

I stole the mock testing approach from https://github.com/mattermost/matterbuild/pull/28. All credit goes to @ali-farooq0. This will allow much easier testing in the future.

#### Ticket Link
Re-submission of https://github.com/mattermost/mattermost-mattermod/pull/117

### CODE REVIEW

File: server/github_client.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	CreateStatus(ctx context.Context, owner, repo, ref string, status *github.RepoStatus) (*github.RepoStatus, *github.Response, error)
+	Get(ctx context.Context, owner, repo string) (*github.Repository, *github.Response, error)
+	GetCombinedStatus(ctx context.Context, owner, repo, ref string, opts *github.ListOptions) (*github.CombinedStatus, *github.Response, error)
+	ListTeams(ctx context.Context, owner string, repo string, opts *github.ListOptions) ([]*github.Team, *github.Response, error)
+	ListStatuses(ctx context.Context, owner, repo, ref string, opts *github.ListOptions) ([]*github.RepoStatus, *github.Response, error)
+}
+
+// GithubClient wraps the github.Client with relevant interfaces.
+type GithubClient struct {
+	*github.Client
```
Comment (hanzei) [inline server/github_client.go:62]: @metanerd This is the fix for the panic

File: server/pull_request.go [hunk 2]
```diff
@@ -457,6 +468,44 @@ func (s *Server) CleanOutdatedPRs() {
 	mlog.Info("Finished update the outdated prs in the mattermod database....")
 }
 
+func (s *Server) CleanUpLabels(client *GithubClient, pr *model.PullRequest) {
+	labelsToRemove := []string{
```
Comment (metanerd) [inline server/pull_request.go:472]: Could we move these to the config please?

Comment (hanzei) [reply server/pull_request.go]: Done in e80b619cdbd873934459ac8284b547cb809c38ac

**Comment by hanzei** [comment]
@ali-farooq0 Correct

**Pull request merged**